src/coupled_array.jl

```diff
@@ -0,0 +1,47 @@
+type CoupledArray{T,T2} <: AbstractArray{Float64,1}
+  u::T
+  u_control::T2
+  order::Bool
+end
+
+Base.length(A::CoupledArray) = length(A.u) + length(A.u_control)
+Base.size(A::CoupledArray) = (length(A),)
+function Base.getindex(A::CoupledArray,i::Int)
+  if A.order == true
+    i <= length(A.u) ? A.u[i] : A.u_control[i-length(A.u)]
+  else
+    i <= length(A.u) ? A.u_control[i] : A.u[i-length(A.u)]
+  end
+end
+
+function Base.getindex(A::CoupledArray,I...)
+  A[sub2ind(A.u,I...)]
+end
+
+function Base.getindex(A::CoupledArray,I::CartesianIndex{1})
```

Comment:
Not that I'm aware of, but it was implemented for `ExtendedJumpArray` so I put it in there to be safe.